handler/src/main/java/io/netty/handler/pcap/PcapWriteHandler.java

```diff
@@ -617,6 +597,92 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
         ctx.fireExceptionCaught(cause);
     }
 
+    /**
+     * Logger for TCP
+     */
+    private void logTCP(boolean isWriteOperation, int bytes, int sendSegmentNumber, int receiveSegmentNumber,
+                        InetSocketAddress srcAddr, InetSocketAddress dstAddr, boolean ackOnly) {
+        // If `ackOnly` is `true` when we don't need to write any data so we'll not
+        // log number of bytes being written and mark the operation as "TCP ACK".
+        if (logger.isDebugEnabled()) {
+            if (ackOnly) {
+                logger.debug("Writing TCP ACK, isWriteOperation {}, Segment Number {}, Ack Number {}, Src Addr {}, "
+                        + "Dst Addr {}", isWriteOperation, sendSegmentNumber, receiveSegmentNumber, dstAddr, srcAddr);
+            } else {
+                logger.debug("Writing TCP Data of {} Bytes, isWriteOperation {}, Segment Number {}, Ack Number {}, " +
+                                "Src Addr {}, Dst Addr {}", bytes, isWriteOperation, sendSegmentNumber,
+                        receiveSegmentNumber, srcAddr, dstAddr);
+            }
+        }
+    }
+
+    OutputStream outputStream() {
+        return outputStream;
+    }
+
+    boolean sharedOutputStream() {
+        return sharedOutputStream;
+    }
+
+    public State state() {
+        return state;
+    }
+
+    /**
+     * Pause the {@link PcapWriteHandler} from writing packets to the {@link OutputStream}.
+     */
+    public void pause() {
+        if (state == State.STARTED) {
```

Comment:
Should we make `state` an `AtomicReference` as these methods can be called from multiple threads  and use `compareAndSet(...)`?